### ISSUE TITLE
Lava in overworld

### ISSUE THREAD

**Issue by SirStudMuffin** [MyM-ModpackTeam/hypovolemia#207]
http://imgur.com/Qw1cKHJ

**Comment by TheIcyOne** [comment]
Does it generate anywhere other than in ravines?

**Comment by SirStudMuffin** [comment]
http://imgur.com/cvPpTga

**Comment by TheIcyOne** [comment]
Looks like all underground lava still generates, but nothing found above ground.

**Comment by Slind14** [comment]
can't be prevented with biome tweaker would need a custom mod unless one exists already

**Comment by TheIcyOne** [comment]
Lava doesn't generate in the nether, at least.

**Comment by Slind14** [comment]
hmm, I remember a discussion with the biome tweaks dev regarding the lava and that isn't possible with biome tweaker due to the way it generates. but that was more than half a year ago.

**Comment by drz2k** [comment]
It would be very heavy (on the already heavy) worldgen to improve this. So i found a settings that was okay for me in terms of heavy worldgen and left lava blocks.
It works this way : BT checks for x times in each chunk if block xyz is lava, then replaces it. You COULD set this to something like 5.000.000 but the worlgend would become way too heavy.
Thoo the lava lake in the screenshot above looks a bit "much" compared to my original worldgen.

**Comment by TheIcyOne** [comment]
That looks like vanilla's underground lava generation, which should be cancellable using Forge's event bus system.

**Comment by kindlich** [comment]
Reika made a mod for changing cave and underground dungeon gen. 

It was called "cave control"



https://sites.google.com/site/reikasminecraft/cave-control maybe we should add either this or an other mod if lava shoudnt spawn in the overworld it could also help with the mineshaft issue

**Comment by johnfg12** [comment]
how do we get hold of reika?

**Comment by Slind14** [comment]
might be worth doing a benchmark first, if it doubles worldgen time or even worse, probably best not to use it.

**Comment by bookerthegeek** [comment]
I can check biome tweaker, it should be doable.  Cave control requires dragon API @Slind14 so I don't know if that would work. If it would it will work. 

I'll check biome tweaker tonight.

**Comment by bookerthegeek** [comment]
Cave control works perfectly. Also solves https://github.com/MyM-ModpackTeam/hypovolemia/issues/216

I'll go ask for perms if you want to use it. We can also use it to disable Dungeons and strongholds. 

(This requires Dragon API..... so no cauldron unless someone can work a miracle.)

[Before](http://goo.gl/TSOTLX) & [After](http://goo.gl/SYUnY3)

**Comment by johnfg12** [comment]
Doesn't work with kcauldron?

**Comment by bookerthegeek** [comment]
Not really. But we could add it anyways.

- Client side won't matter as clients with it can connect to servers without it.
 - Will work for singleplayer games and gen the world correctly.
- Server side, Worlds can be Generated on Forge, then disable the mod(s) after all the Pregen then switch the server to Cauldron.
 - We don't need it after the world is generated.

**Comment by Slind14** [comment]
MrWisski has been using Dragon API v14 on a cauldron server without issues, if it is only v15 and those features existed back in v14, we could use that.

**Comment by bookerthegeek** [comment]
They exist on both.  I'll go ask for permissions to include it in the pack.

**Comment by johnfg12** [comment]
thanks booker :)

**Comment by SirStudMuffin** [comment]
Pi said he could do it with 3 lines of code

**Comment by bookerthegeek** [comment]
True.  But this will allow us the possibility of adding other mods such as expanded Redstone, and void monster. :)

I think I'll talk to Pi tonight about the three lines of code.

**Comment by johnfg12** [comment]
expanded Redstone is nice i was reading his what not to say section  earlier so funny :P

**Comment by bookerthegeek** [comment]
Right?

**Comment by johnfg12** [comment]
any update on getting perms?

**Comment by johnfg12** [comment]
https://sites.google.com/site/reikasminecraft/approved-modpacks im guessing we got perms then?

**Comment by tyler489** [comment]
Yeah. Ask if we can change recipys if we add a disclaimer tooltip

**Comment by johnfg12** [comment]
we only need cave control atm

**Comment by johnfg12** [comment]
where can we get version  14 from?

**Comment by knoxz** [comment]
you could use my mod and replace all the lava source blocks with blood or anything else. you would end up with close to no lava.
Cant retrogen that though ;-)

**Comment by bookerthegeek** [comment]
http://www.minecraftforum.net/forums/mapping-and-modding/minecraft-mods/1291655-reikas-mods-tech-worldgen-civilization-and-more?page=1391#c28190

Permission Granted.

**Comment by tyler489** [comment]
What mod @knoxz

**Comment by Bennyboy1695** [comment]
@tyler489 http://minecraft.curseforge.com/projects/ore-gen-anything
